src/OpeningHours.php

```diff
@@ -159,4 +171,13 @@ protected function normalizeDayName(string $day)
 
         return $day;
     }
+
+    protected function setTimezone(DateTimeInterface $date)
+    {
+        if ($this->timezone) {
+            $date->setTimezone($this->timezone);
```

Comment:
this is a bug.
the only classes that can directly implements `DateTimeInterface` are `DateTimeImmutable` and `DateTime`. Both classes have a `setTimeZone` method but since they differs, one is immutable and the other is not you should update your code like this

```php
<?php
$data = $date->setTimezone($this->timezone);
```
otherwise your code won't work with `DateTimeImmutable` objects or object extending  `DateTimeImmutable`